executors/clipseg/executor.py

```diff
@@ -0,0 +1,214 @@
+import enum
+import os
+import sys
+import shutil
+import time
+
+from PIL import Image, ImageOps
+from io import BytesIO
+from pathlib import Path
+from typing import Dict
+from urllib.request import urlopen
+
+import cv2
+import numpy as np
+import torch
+
+from clipseg.clipseg import CLIPDensePredT
+from jina import Executor, DocumentArray, Document, requests
+from torchvision import transforms
+
+
+class THRESHOLDING_METHODS(str, enum.Enum):
+    NONE = 'none' # Do not threshold
+    BINARY = 'binary'
+    ADAPTIVE_MEAN = 'adaptive_mean'
+    ADAPTIVE_GAUSSIAN = 'adaptive_gaussian'
+
+
+THRESHOLD_ADAPTIVE_DEFAULT_BLOCK_SIZE = 11
+THRESHOLD_ADAPTIVE_DEFAULT_C = 2.
+THRESHOLD_BINARY_DEFAULT_STRENGTH_VALUE = 85
+WEIGHTS_LOC = 'https://owncloud.gwdg.de/index.php/s/ioHbRzFx6th32hn/download'
```

Comment:
Lets put this argument in the init function of the Executor with default value.

So that someone can change it if he want to use other weight